python/cudf/cudf/core/index.py

```diff
@@ -2078,6 +2078,54 @@ def quarter(self):
         res = extract_quarter(self._values)
         return Index(res, dtype="int8")
 
+    @_cudf_nvtx_annotate
+    def day_name(self, locale=None):
+        """
+        Return the day names. Currently supports English locale only.
+
+        Examples
+        --------
+        >>> import pandas as pd
+        >>> import cudf
+        >>> datetime_index = cudf.Index(pd.date_range("2016-12-31",
+        ...     "2017-01-08", freq="D"))
+        >>> datetime_index
+        DatetimeIndex(['2016-12-31', '2017-01-01', '2017-01-02', '2017-01-03',
+                    '2017-01-04', '2017-01-05', '2017-01-06', '2017-01-07',
+                    '2017-01-08'],
+                    dtype='datetime64[ns]')
+        >>> datetime_index.day_name()
+        Index(['Saturday', 'Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday',
+               'Friday', 'Saturday'], dtype='object')
+        """
+        days = self._get_dt_field("weekday")
+        day_names = cudf.core.tools.datetimes._get_date_name_field(
```

Comment:
I think it might be better to cast the column to string and use the column's `find_and replace`